RecipeApp/AuthorizationViewController/AuthorizationViewController.swift

```diff
@@ -0,0 +1,466 @@
+        activityIndicator.startAnimating()
+        loginButton.addSubview(activityIndicator)
+
+        /// таймер на 3 секунды
+        Timer.scheduledTimer(withTimeInterval: 3.0, repeats: false) { _ in
+            /// остановка анимации и скрытие ее
+            activityIndicator.stopAnimating()
+            activityIndicator.removeFromSuperview()
+
+            self.presenter?.loginButtonTapped(
+                login: self.emailTextField.text ?? "",
+                password: self.passwordTextField.text ?? ""
+            )
+        }
+    }
+
+    /// метод отработки появление плашки для презентора
+    func updateLoginButton(result: Bool) {
+        if result == false {
+            errorView.isHidden = false
+            errorMessageLabel.isHidden = false
+        } else {
+            errorView.isHidden = false
+            errorMessageLabel.isHidden = false
+        }
+    }
+}
+
+// MARK: - ViewController: UITextFieldDelegate
+
+/// Расширение для управления текстовыми полями
+extension AuthorizationViewController: UITextFieldDelegate {
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if textField == emailTextField {
+            presenter?.validateEmail(email: textField.text ?? "")
+        } else if textField == passwordTextField {
+            presenter?.validatePassword(password: textField.text ?? "")
+        }
+    }
+
+    @objc private func doneButtonTapped() {
+        view.endEditing(true)
+    }
+}
+
+// MARK: - Keyboard methods
+
+extension AuthorizationViewController {
+    func setupKeyboard() {
+        setupInputAccessoryView()
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShow(notification:)),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(notification:)),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    /// Обработчик появления клавиатуры
+    @objc func keyboardWillShow(notification: Notification) {
+        guard let keyboardFrame = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?
+            .cgRectValue else { return }
+
+        let keyboardHeight = keyboardFrame.height
+        let distanceBetweenButtonAndKeyboard: CGFloat = 10
+        let buttonYPosition = view.frame.height - keyboardHeight - loginButton.frame
+            .height - distanceBetweenButtonAndKeyboard
+
+        UIView.animate(withDuration: 0.3) {
+            self.loginButton.frame.origin.y = buttonYPosition
+        }
+    }
+
+    /// обработчик скрытия клавиатуры
+    @objc func keyboardWillHide(notification: Notification) {
+        UIView.animate(withDuration: 0.3) {
+            self.loginButton.frame.origin.y = self.view.frame.height - 45 - 20
```

Comment:
```suggestion
            self.loginButton.frame.origin.y = self.view.frame.height - 65
```
? 65 в константы - что за магическое число?